core/station/impl/src/models/request_policy_rule.rs

```diff
@@ -413,14 +417,27 @@ impl
                             });
                         }
                         Ok(account) => {
-                            let is_in_address_book = ADDRESS_BOOK_REPOSITORY
-                                .exists(account.blockchain, transfer.input.to.clone());
-
-                            if is_in_address_book {
-                                return Ok(RequestPolicyRuleResult {
-                                    status: EvaluationStatus::Approved,
-                                    evaluated_rule: EvaluatedRequestPolicyRule::AllowListed,
-                                });
+                            for account_asset in account.assets {
+                                let Some(asset) = ASSET_REPOSITORY.get(&account_asset.asset_id)
+                                else {
+                                    print(format!(
+                                        "Asset `{}` not found in account `{}`.",
+                                        Uuid::from_bytes(account_asset.asset_id).hyphenated(),
+                                        Uuid::from_bytes(account.id).hyphenated()
+                                    ));
+
+                                    continue;
```

Comment:
This loops over the assets of an account. If an asset is not found, it should still try to check the other assets, shouldn't it? If this was the only asset, or the address is not in the address book for the other assets, it will fail. Should it fail automatically if any of the assets are missing?

Reply:
ah right, makes sense then to not fail here, and in the case of this asset not being found it would already be rejected later ([here](https://github.com/dfinity/orbit/pull/374/files#diff-becb38dd3888e9e5d333d20aab853c0108febf3a6fbd80b379dfdbc56274169fR435)) which would be the expected flow